server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/SieveScriptRoutes.java

```diff
@@ -126,30 +127,29 @@ public void defineAddActiveSieveScript(Service service) {
     private HaltException addActiveSieveScript(Request request, Response response) throws UsersRepositoryException, QuotaExceededException, StorageException, ScriptNotFoundException {
         User user = extractUser(request);
         ScriptName script = extractScriptName(request);
+        boolean isActivated = isActivated(request.queryParams(ACTIVATE_PARAMS));
         sieveRepository.putScript(user, script, extractSieveScriptFromRequest(request));
-        if (isActivated(request.queryParams(ACTIVATE_PARAMS))) {
+        if (isActivated) {
             sieveRepository.setActive(user, script);
         }
         return halt(HttpStatus.NO_CONTENT_204);
     }
 
-    private User extractUser(Request request) throws UsersRepositoryException {
+    private User extractUser(Request request) {
         Optional<String> userName = Optional.ofNullable(request.params(USER_NAME));
-        if (!userName.isPresent() || Strings.isNullOrEmpty(userName.get().trim())) {
-            throw400withInvalidArgument("Invalid username");
-        }
-
-        if (!usersRepository.contains(userName.get())) {
-            throw404("User not exist");
-        }
-
-        return User.fromUsername(userName.get());
+        userName.map(String::trim)
+            .filter(StringUtils::isNotEmpty)
+            .orElseThrow(() -> throw400withInvalidArgument("Invalid username"));
+        return userName.map(String::trim)
+            .filter(Throwing.predicate(name -> usersRepository.contains(name)))
+            .map(User::fromUsername)
+            .orElseThrow(() -> throw404("User not found"));
```

Comment:
Technically the code above relyes on a side effect (the fact that the terminal call orElseThow consume the optional)

I'd see more something like

```
String userName = Optional.ofNullable(request.params(USER_NAME))
            .map(String::trim)
            .filter(StringUtils::isNotEmpty)
            .orElseThrow(() -> throw400withInvalidArgument("Invalid username"));

if (usersRepository.contains(username)) {
    return User.fromUserName(username);
}
throw404("User not found");
```